backbone.js

```diff
@@ -38,6 +38,13 @@
   var array = [];
   var slice = array.slice;
 
+  // Optimized deep comparison function.
+  var _isEqual = _.isEqual;
+
+  function isEqual( a, b ){
+      return a && b && typeof a === 'object' && typeof b === 'object' ? _isEqual( a, b ) : a === b;
```

Comment:
Also, NaN is not serializable to JSON, so it's not an issue for Models. Although, it won't hit issues with NaN either. NaN !== NaN, so what. Bound case your application isn't supposed to work in either, what's the deal if you'll receive extra change event.